donate/payments/tasks.py

```diff
@@ -331,6 +341,71 @@ def process_charge_failed(event):
         })
 
 
+class MigrateStripeSubscription:
+    def process(self, charge, subscription):
+        # Grab the payment card ID from the subscription's payment source list (there will only be one)
+        card_id = subscription.customer.sources.data[0].id
+
+        # Confirm the payment method was migrated to Braintree by looking up
+        # BT Payment Methods using the Stripe card ID
+        payment_method = gateway.payment_method.find(card_id)
+
+        # Ensure we found a Payment Method, and if not, report it to Sentry as an error
+        if not payment_method:
+            message = f'The payment method was not migrated to' \
+                      f' Braintree for this charge: {charge.id}'
+            print(message)
+            logger.error(message, exc_info=True)
+            return
+
+        # Determine the Braintree plan, merchant account, and price
+        # for the subscription based on the subscription's attributes
+        BT_plan_id = get_plan_id(charge.currency)
+        BT_merchant_account = get_merchant_account_id_for_card(charge.currency)
+        BT_price = Decimal(zero_decimal_currency_fix(charge.amount, charge.currency))
+        BT_first_billing_date = datetime.utcfromtimestamp(subscription.current_period_end)
+
+        # Create the Braintree subscription and record the result for analysis
+        BT_create_subscription_result = gateway.subscription.create({
+            'payment_method_token': payment_method.token,
```

Comment:
Do we need to verify that `payment_method.token` exists, where we're checking if `payment_method` exists? Or does this exist by definition and we don't need to worry about this prop possibly not existing?

Reply:
Token must exist of there is a payment method object. 👍